File: java/resistor-color-duo/ResistorColorDuo.java
import java.util.Map;

/**
 * Resistor Color Duo
 * https://exercism.org/tracks/java/exercises/resistor-color-duo
 */
class ResistorColorDuo {

    private static final Map<String, Integer> ENCODING = Map.of(
        "black",  0,
        "brown",  1,
        "red",    2,
        "orange", 3,
        "yellow", 4,
        "green",  5,
        "blue",   6,
        "violet", 7,
        "grey",   8,
        "white",  9
    );

    int value(String[] colors) {
        return 10 * ENCODING.get(colors[0]) + ENCODING.get(colors[1]);
    }
}

File: kotlin/ResistorColorDuo.kt
/**
 * https://exercism.io/my/tracks/kotlin
 * Resistor Color Duo
 */

enum class Color(val code: Int) {
    BLACK(0),
    BROWN(1),
    RED(2),
    ORANGE(3),
    YELLOW(4),
    GREEN(5),
    BLUE(6),
    VIOLET(7),
    GREY(8),
    WHITE(9);
}

object ResistorColorDuo {

    fun value(vararg colors: Color): Int {
        if (colors.size < 2) {
            throw IllegalArgumentException("Provide at least 2 colors")
        }

        return colors[0].code * 10 + colors[1].code
    }
}
